Describe duplication of state machines for post-handshake messages

hanno-arm opened the pull request:
As discussed in https://mailarchive.ietf.org/arch/msg/tls/Pvg2IUJAYTyQDygUBDV_D_ozRHM/, post-handshake messages should be handled via independent state machines. This PR is a first attempt to add a section to the spec explicitly mentioning this.

thomas-fossati commented on draft-ietf-tls-dtls13.md:
```diff
@@ -1349,6 +1349,38 @@ reset to the initial value.  After a long period of idleness, no less
+5. Post-handshake client authentication
+
+Messages of each category can be sent independently, and reliability is established
+via independent state machines each of which behaves as described in {state-machine}.
```
```suggestion
via independent state machines each of which behaves as described in {{state-machine}}.
```

hanno-arm replied:
Applied, thanks! :+1:

thomas-fossati commented on draft-ietf-tls-dtls13.md:
```diff
@@ -1642,11 +1674,11 @@ of the NewSessionTicket, and KeyUpdate messages.
-implicitly acknowledge that message. Implementations MAY
-acknowledge the records corresponding to each transmission of
+immediately follows the flight. As explained in {{state-machine-duplication}},
+each category of post-handshake messages (e.g. NewSessionTicket or KeyUpdate)
```
```suggestion
each category of post-handshake messages (e.g., NewSessionTicket or KeyUpdate)
```

hanno-arm replied:
Applied, thanks! :+1:

hanno-arm commented:
@thomas-fossati I have fixed two further typos, and also added further explanations on when multiple messages of a given post-HS category may be sent. Can you please take another look?

ekr commented on draft-ietf-tls-dtls13.md:
```diff
@@ -1349,6 +1349,49 @@ reset to the initial value.  After a long period of idleness, no less
+of the same type has not yet been acknowledged.
+
+Note: Except for post-handshake client authentication, which involves handshake messages
+in both directions, post-handshake messages are single-flight, and their respective state
```
unidirectional.

ekr commented:
Merged by hand
